src/tools/schema2kotlin.ts

```diff
@@ -72,12 +79,27 @@ import arcs.sdk.utf8ToString
     const handleDecls: string[] = [];
     for (const connection of particle.connections) {
       const handleName = connection.name;
-      const handleType = connection.type.isCollectionType() ? 'Collection' : 'Singleton';
       const entityType = `${particleName}_${this.upperFirst(connection.name)}`;
-      handleDecls.push(`protected val ${handleName} = ${handleType}(this, "${handleName}", ${entityType}_Spec())`);
+      let handleConcreteType = connection.type.isCollectionType() ? 'Collection' : 'Singleton';
+      let handleInterfaceType: string;
+      switch (connection.direction) {
+        case 'reads writes':
```

Comment:
Could it ever be `writes reads`?

Reply:
Nope, only 'reads writes'. The cases are actually known and type checked by the compiler.